cluster-autoscaler/Makefile

```diff
@@ -9,6 +9,7 @@ GOOS?=linux
 GOARCH?=$(shell go env GOARCH)
 REGISTRY?=staging-k8s.gcr.io
 DOCKER_NETWORK?=default
+SUPPORTED_BUILD_TAGS=gce aws azure kubemark alicloud magnum digitalocean clusterapi huaweicloud ionoscloud linode hetzner bizflycloud brightbox equinixmetal oci vultr tencentcloud scaleway externalgrpc civo rancher volcengine baiducloud cherry cloudstack exoscale kamatera ovhcloud
```

Comment:
I wonder if we could have a single source of truth for this. I thought something like:

```
ls cluster-autoscaler/cloudprovider/builder/ | grep -e '^builder_.*\.go' | sed 's/builder_\(.*\)\.go/\1/'
```

Though the downside is if the list becomes empty, the test will keep on passing. Thoughts?

Reply:
I incorporated this improvement and ensured that there are the right error paths as well, ptal. Thanks!